src/TransactionController.ts

```diff
@@ -413,6 +421,105 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
 		const newTransactions = this.state.transactions.filter(({ networkID }) => networkID !== currentNetworkID);
 		this.update({ transactions: newTransactions });
 	}
+
+	/**
+	 * Gets all transactions from etherscan for a specific address
+	 * optionally starting from a specific block
+	 *
+	 * @param address - string representing the address to fetch the transactions from
+	 * @param fromBlock - string representing the block number (optional)
+	 * @returns - Promise resolving to an string containing the block number of the latest incoming transaction.
+	 */
+	async fetchAll(address: string, fromBlock?: string) {
+		const network = this.context.NetworkController;
+		const currentNetworkID = network.state.network;
+		const apiUrl = this._getEtherscanApiUrl(currentNetworkID);
+		/* istanbul ignore next */
+		if (!apiUrl) {
+			return null;
+		}
+		let url = `${apiUrl}/api?module=account&action=txlist&address=${address}&tag=latest&page=1`;
+		/* istanbul ignore next */
+		if (fromBlock) {
+			url += `&startBlock=${fromBlock}`;
+		}
+		const response = await fetch(url);
+		const parsedResponse = await response.json();
+		/* istanbul ignore next */
+		if (parsedResponse.status !== '0' && parsedResponse.result.length > 0) {
+			const remoteTxList: any = {};
+			const remoteTxs = parsedResponse.result.map((tx: any) => {
+				remoteTxList[tx.hash] = 1;
+				return this._normalizeTxFromEtherscan(tx, currentNetworkID);
+			});
+
+			const localTxs = this.state.transactions.filter(
+				(tx: TransactionMeta) => !remoteTxList[`${tx.transactionHash}`]
+			);
+
+			const allTxs = [...remoteTxs, ...localTxs];
+			allTxs.sort((a, b) => (a.time < b.time ? -1 : 1));
+
+			let latestIncomingTxBlockNumber: any = null;
+			allTxs.forEach((tx) => {
+				if (tx.transaction.to.toLowerCase() === address.toLowerCase()) {
+					if (
+						!latestIncomingTxBlockNumber ||
+						parseInt(latestIncomingTxBlockNumber, 10) < parseInt(tx.blockNumber, 10)
+					) {
+						latestIncomingTxBlockNumber = tx.blockNumber;
+					}
+				}
+			});
+
+			this.update({ transactions: allTxs });
+			return latestIncomingTxBlockNumber;
+		}
+		/* istanbul ignore next */
+		return null;
+	}
+
+	/**
+	 * Normalizes the transaction information from etherscan
+	 * to be compatible with the TransactionMeta interface
+	 *
+	 * @param txMeta - Object containing the transaction information
+	 * @param currentNetworkID - string representing the current network id
+	 * @returns - TransactionMeta
+	 */
+	_normalizeTxFromEtherscan(txMeta: any, currentNetworkID: string) {
```

Comment:
is this a private method? if yes, we may want to make it `private` at the beginning of the controller definition to maintain the coding style and reduce the API